File: Misc DMOJ/factorial/src/factorial.java
import java.util.*;
import java.io.*;

public class factorial {
	/* 
	 * Copy-pasting code is NOT cool! Please do not copy and paste my code as a submission to DMOJ.
	 * github.com/jdabtieu/competitive-programming
	 */
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	
	static long table[] = {1L, 1L, 2L, 6L, 24L, 120L, 720L, 5040L, 40320L, 362880L, 3628800L, 39916800L, 479001600L, 1932053504L, 1278945280L, 2004310016L, 2004189184L, 4006445056L, 3396534272L, 109641728L, 2192834560L, 3099852800L, 3772252160L, 862453760L, 3519021056L, 2076180480L, 2441084928L, 1484783616L, 2919235584L, 3053453312L, 1409286144L, 738197504L, 2147483648L, 2147483648L};

	public static void main(String[] args) throws IOException {
		for (int cases = Integer.parseInt(br.readLine()); cases > 0; cases--) {
		    String s = br.readLine();
		    if (s.length() > 2 || Integer.parseInt(s) >= 34) {
		        System.out.println(0);
		        continue;
		    }
			System.out.println(table[Integer.parseInt(s)]);
		}
	}
}

File: Misc DMOJ/factorial/src/factorial.cpp
#include <bits/stdc++.h>

#define uint_64 unsigned long long

using namespace std;

/*
 * Copy-pasting code is NOT cool! Please do not copy and paste my code as a submission to DMOJ.
 * github.com/jdabtieu/competitive-programming
 */
int main() {
    uint_64 table[34] = {1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880, 3628800, 39916800, 479001600, 1932053504, 1278945280, 2004310016, 2004189184, 4006445056, 3396534272, 109641728, 2192834560, 3099852800, 3772252160, 862453760, 3519021056, 2076180480, 2441084928, 1484783616, 2919235584, 3053453312, 1409286144, 738197504, 2147483648, 2147483648};
    int cases; scanf("%d", &cases);
    while (cases--) {
        uint_64 num; scanf("%llu", &num);
        if (num >= 34) {
            printf("0\n");
            continue;
        }
        printf("%llu\n", table[num]);
    }
}